File: fs/src/main/scala/farjs/fs/popups/FolderShortcutsController.scala
package farjs.fs.popups

import scommons.react._

object FolderShortcutsController extends FunctionComponent[FolderShortcutsControllerProps] {

  private[popups] var folderShortcutsPopup: UiComponent[FolderShortcutsPopupProps] = FolderShortcutsPopup

  protected def render(compProps: Props): ReactElement = {
    val props = compProps.plain

    if (props.showPopup) {
      <(folderShortcutsPopup())(^.wrapped := FolderShortcutsPopupProps(
        onChangeDir = { dir =>
          props.onClose()
          props.onChangeDir(dir)
        },
        onClose = props.onClose
      ))()
    }
    else null
  }
}

File: fs/popups/FolderShortcutsController.mjs
import React from "react";
import FolderShortcutsPopup from "./FolderShortcutsPopup.mjs";

const h = React.createElement;

/**
 * @typedef {{
 *  readonly showPopup: boolean;
 *  onChangeDir(dir: string): void;
 *  onClose(): void;
 * }} FolderShortcutsControllerProps
 */

/**
 * @param {FolderShortcutsControllerProps} props
 */
const FolderShortcutsController = (props) => {
  const { folderShortcutsPopup } = FolderShortcutsController;

  return props.showPopup
    ? h(folderShortcutsPopup, {
        onChangeDir: (dir) => {
          props.onClose();
          props.onChangeDir(dir);
        },
        onClose: props.onClose,
      })
    : null;
};

FolderShortcutsController.displayName = "FolderShortcutsController";
FolderShortcutsController.folderShortcutsPopup = FolderShortcutsPopup;

export default FolderShortcutsController;
